AISKU/Tests/perfTests.ts

```diff
@@ -0,0 +1,363 @@
+  Event,
+  Trace,
+  Exception,
+  Metric,
+  PageView,
+  PageViewPerformance,
+  RemoteDependencyData,
+} from "@microsoft/applicationinsights-common";
+import { EventValidator } from "./TelemetryValidation/EventValidator";
+import { TraceValidator } from "./TelemetryValidation/TraceValidator";
+import { ExceptionValidator } from "./TelemetryValidation/ExceptionValidator";
+import { MetricValidator } from "./TelemetryValidation/MetricValidator";
+import { PageViewPerformanceValidator } from "./TelemetryValidation/PageViewPerformanceValidator";
+import { PageViewValidator } from "./TelemetryValidation/PageViewValidator";
+import { RemoteDepdencyValidator } from "./TelemetryValidation/RemoteDepdencyValidator";
+
+export class PerfTests extends TestClass {
+  private readonly _instrumentationKey = "b7170927-2d1c-44f1-acec-59f4e1751c11";
+
+  private _ai: IApplicationInsights;
+
+  // Sinon
+  private errorSpy: SinonSpy;
+  private successSpy: SinonSpy;
+  private loggingSpy: SinonSpy;
+
+  private delay = 100;
+  private _notificationManager;
+  private _sendNotifications;
+
+  public testInitialize() {
+    try {
+      this.useFakeServer = false;
+      (sinon.fakeServer as any).restore();
+      this.useFakeTimers = false;
+      this.clock.restore();
+
+      // Setup Notification Listener
+      this._sendNotifications = [];
+      this._notificationManager = new NotificationManager();
+      this._notificationManager.addNotificationListener({
+        perfEvent: (perfEvent: IPerfEvent): void => {
+          this._sendNotifications.push(perfEvent);
+        },
+      });
+
+      const init = new ApplicationInsights({
+        config: {
+          instrumentationKey: this._instrumentationKey,
+          extensionConfig: {
+            AppInsightsChannelPlugin: {
+              maxBatchInterval: 2000,
+              maxBatchSizeInBytes: 10 * 1024 * 1024, // 10 MB
+            },
+          },
+        },
+      });
+
+      init.core.setPerfMgr(new PerfManager(this._notificationManager));
+      init.loadAppInsights(false, null, this._notificationManager);
+      this._ai = init;
+
+      // Setup Sinon stuff
+      const sender: Sender = this._ai.appInsights.core.getTransmissionControls()[0][0] as Sender;
+      this.errorSpy = this.sandbox.spy(sender, "_onError");
+      this.successSpy = this.sandbox.spy(sender, "_onSuccess");
+      this.loggingSpy = this.sandbox.stub(
+        this._ai.appInsights.core.logger,
+        "throwInternal"
+      );
+    } catch (e) {
+      console.error("Failed to initialize");
+    }
+  }
+
+  public testCleanup() {
+    this.useFakeServer = true;
+    this.useFakeTimers = true;
+    this._sendNotifications = [];
+    this.successSpy.restore();
+  }
+
+  public registerTests() {
+    this.addPerfTests();
+  }
+
+  private addPerfTests(): void {
+    this.testCase({
+      name: "AISKU Perf Tests -> load AppInsights",
+      test: () => {
+        QUnit.assert.ok(this._sendNotifications.length === 1);
```

Comment:
This is what i mean by my comment above, this is testing the side effect of the testIntialize(), if some other test wants or needs to change the initialization it could break this.

solution: each test should setup and trigger the conditions it's testing.
We can still have common object creation and clearing out global values in testinitialize() though